relayer/proxy/proxy.go

```diff
@@ -137,7 +139,25 @@ func (proxy *Proxy) signResponse(relayResponse *servicerTypes.RelayResponse) (si
 }
 
 func validateSessionRequest(session *sessionTypes.Session, relayRequest *servicerTypes.RelayRequest) error {
-	// TODO: validate relayRequest signature
+	// TODO: cache ring for application at start of session
+	// TODO: add ring comparison (https://github.com/noot/ring-go/issues/13)
+
+	sigBz, err := relayRequest.GetSignableBytes()
+	if err != nil {
+		return fmt.Errorf("failed to get request's signable bytes: %w", err)
+	}
+
+	ringSig := new(ring.RingSig)
+	crv := ring_secp256k1.NewCurve()
+	if err := ringSig.Deserialize(crv, relayRequest.Signature); err != nil {
```

Comment:
Not sure what this is doing... Does this mean that `ringSig` is stateful?

Reply:
The `RingSig` struct contains the ring and public keys and this is needed for verification. This serialisation/deserialisation step enables us to transport the ring sig for verification.

```go
// RingSig represents a ring signature.
type RingSig struct {
	ring  *Ring          // array of public keys
	c     types.Scalar   // ring signature challenge
	s     []types.Scalar // ring signature values
	image types.Point    // key image
}
```